src/librustc_borrowck/diagnostics.rs

```diff
@@ -314,6 +314,59 @@ let c = &i; // still ok!
 ```
 "##,
 
+E0504: r##"
+This error occurs when an attempt is made to move a borrowed variable into a
+closure. For example:
+
+```compile_fail
```

Comment:
IMO we should have both examples. This error can crop up a lot in threaded situations, and an extra example addressing it specifically would be nice.